azext_iot/tests/iothub/devices/test_hub_export_import_int.py

```diff
@@ -0,0 +1,506 @@
+import random
+import pytest
+import json
+import os
+import time
+from pathlib import Path
+from azext_iot.common.shared import DeviceAuthApiType
+
+from azext_iot.tests.settings import DynamoSettings, ENV_SET_TEST_IOTHUB_REQUIRED, ENV_SET_TEST_IOTHUB_OPTIONAL
+from azext_iot.tests.iothub import IoTLiveScenarioTest
+from azext_iot.tests.test_constants import ResourceTypes
+from azext_iot.tests.generators import generate_generic_id
+from azext_iot.common.utility import generate_key, read_file_content
+from azext_iot.common.certops import create_self_signed_certificate
+from azext_iot.tests.helpers import add_test_tag
+from azext_iot.tests.iothub import (
+    PRIMARY_THUMBPRINT,
+    SECONDARY_THUMBPRINT,
+    DATAPLANE_AUTH_TYPES,
+    DEVICE_TYPES,
+)
+
+settings = DynamoSettings(req_env_set=ENV_SET_TEST_IOTHUB_REQUIRED, opt_env_set=ENV_SET_TEST_IOTHUB_OPTIONAL)
+CWD = os.path.dirname(os.path.abspath(__file__))
+
+
+class TestHubExportImport(IoTLiveScenarioTest):
+    def __init__(self, test_case):
+        super(TestHubExportImport, self).__init__(test_case)
+        self.dest_hub = settings.env.azext_iot_desthub or "test-hub-" + generate_generic_id()
+        self.dest_hub_rg = settings.env.azext_iot_destrg or settings.env.azext_iot_testrg
+
+        # create destination hub
+
+        if not settings.env.azext_iot_desthub:
+            self.create_hub(self.dest_hub, self.dest_hub_rg)
+
+        add_test_tag(
+            cmd=self.cmd,
+            name=self.dest_hub,
+            rg=self.dest_hub_rg,
+            rtype=ResourceTypes.hub.value,
+            test_tag=test_case
+        )
+
+        self.dest_hub_cstring = self.cmd(
+            f"iot hub connection-string show -n {self.dest_hub} -g {self.dest_hub_rg}"
+        ).get_output_in_json()["connectionString"]
+
+        self.filename = generate_generic_id() + ".json"
+
+        self.create_hub_state()
+
+    def create_hub_state(self):
```

Comment:
We need to test more states than just one loaded state.  Pytest fixtures and parameterized testing can help iterate through and testing states faster.

Reply:
This might have to be taken care of after I leave. I don't think I'll have time to add in several more testing states.